kie-server-parent/kie-server-controller/kie-server-controller-websocket-common/src/main/java/org/kie/server/controller/websocket/common/WebSocketClientImpl.java

```diff
@@ -124,37 +125,39 @@ public void close() {
     }
 
     @Override
-    public void sendTextWithHandler(final String content, final InternalMessageHandler handler) throws IOException {
+    public void sendTextWithHandler(final String content,
+                                    final Consumer<T> handler) throws IOException {
         if (!session.isOpen()) {
             throw new RuntimeException("No connection to controller");
         }
 
-        LOGGER.debug("Sending test message using Web Socket Session with id: {}", session.getId());
-
-        if(handler != null && messageHandler != null) {
-            messageHandler.addHandler(handler);
+        if(handler != null && this.messageHandler != null) {
+            handler.accept(this.messageHandler);
```

Comment:
what happens if the messageHandler is not accepted?

Reply:
Not sure I got this one, but just using the _FunctionalInterface_ _Consumer_ in here just as a method of callback...

Reply:
I was just wondering if there is any situation that the messageHandler won't be accepted ... maybe just misled by the name of the method :)